lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1794,6 +1836,84 @@ def cmd_sessions(*args)
     true
   end
 
+  def get_matching_sessions(search_term)
+    matching_sessions = {}
+    terms = search_term.split
+    terms.each do |term|
+      matches = filter_sessions_by_search(term)
+      matches.each do |session_id, session|
```

Comment:
am I correct in thinking that this results in your filter being an `or` filter rather than `and`? so if you search for before a certain checkin time and for session to be a meterpreter type rather than getting only meterpreter sessions checking in _before_ a certain time you get _all_ sessions that have checked in _and_ all meterpreter sessions regardless of checkin time?

I would have expected the search to be an `and` and I think that's what users would expect too, but let me know if you disagree

Reply:
I was thinking an or filter for the case of someone searching for multiple different ids or session types, and having it be consistent from there. Seems like a few options -
1) keep it as an `or`, which might lead to some users on the search results if they try to query on multiple fields as you pointed out
2) change it to an `and`, effectively disabling the ability to search for multiple different types of the same field
3) add the logic to make the or/and context sensitive, which might be more convoluted/inconsistent, but has the most functionality
Happy to defer to you here, lmk what you think

Reply:
Let's try and base our implementation off of how docker does it https://docs.docker.com/engine/reference/commandline/images/#filter since thye seem to have a fairly good UX for this kinda thing already
 sessions IDs would be an `or` then anything else would be an `and`
 I really think the ability to provide a before time and an after time to select sessions from a certain range could be a pretty nice feature to have and I think it's more intuitive that way
Probably best to write up the test cases first for what we would expect to happen and maybe get a few more opinions on that when people can see what the workflow will be more easily